### PR TITLE
Componentizes Food Storage

### PR DESCRIPTION
**Pull request by MrMelbert**
## About The Pull Request
Moves the food storage behavior on (formerly) bread, cheese wheels, and cakes from `/food/snacks/store/` to a component.

## Why It's Good For The Game
This lets just about any food/edible thing be used for storing fun things like glass shards, which is pretty neat. Also helps work towards the food refactor.

## Changelog
:cl: Melbert
tweak: You can now remove items from food storage by clicking on them on grab intent.
code: Componentizes food storage from food/snacks/store.
/:cl:

### CODE REVIEW

**Review by ghost — COMMENTED** [review]
Looks good some minor changes needed and then it's perfect. I'll remove the subtype once i get there

File: code/datums/components/storage/food_storage.dm [hunk 1]
```diff
@@ -0,0 +1,182 @@
+/// --Food storage component--
+/// This component lets you slide one item into large foods, such as bread, cheese wheels, or cakes.
+/// Consuming food storages with an item inside can cause unique interactions, such as eating glass shards.
+
+/datum/component/food_storage
```
Comment (Qustinnus) [inline code/datums/components/storage/food_storage.dm:5]: you need to handle the storage food to drop the stored_item if its deleted.

**Comment by LemonInTheDark** [comment]
This ready?

**Comment by MrMelbert** [comment]
I believe so, unless @Qustinnus has any last minute requests/suggestions

**Comment by Qustinnus** [comment]
No, it looks good to me atm,

**Comment by LemonInTheDark** [comment]
I don't understand it, but I will act as if I do for internet clout

**Comment by Qustinnus** [comment]
glad to be maintainer-by-proxy ;^)

**Pull request merged**